<launch>
    <arg name="veh" default="$(env VEHICLE_NAME)" doc="Name of vehicle. ex: megaman"/>
    <arg name="param_file_name" default="default" doc="Specify a param file. ex:megaman." />
    <!-- <arg name="testing" default="false" doc="using for test purposes." /> -->

    <arg name="pkg_name" value="ekf_slam"/>
    <arg name="node_name" default="ekf_slam_node"/>
    <arg name="traffic_mode" doc="Left/Right-hand traffic" default="RHT"/>

    <group ns="$(arg veh)">
        <remap from="~camera_info" to="camera_node/camera_info"/>
        <remap from="~image/compressed" to="camera_node/image/compressed"/>
  	    <remap from="~/left_wheel_encoder_driver_node/tick" to="left_wheel_encoder_driver_node/tick"/>
  	    <remap from="~/right_wheel_encoder_driver_node/tick" to="right_wheel_encoder_driver_node/tick"/>

        <node name="$(arg node_name)" pkg="$(arg pkg_name)" type="$(arg node_name).py" output="screen">
            <rosparam command="load" file="$(find ekf_slam)/config/$(arg node_name)/$(arg param_file_name).yaml"/>
        </node>
    </group>

    <!-- Publications -->

    <!-- Subscriptions -->


</launch>
